pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -461,4 +571,66 @@ static inline uint32_t _update_timeout(uint32_t start, uint32_t timeout)
     return (diff > timeout) ? 0: timeout - diff;
 }
 
+#ifdef SOCK_HAS_ASYNC
+void _udp_cb(sock_udp_t *udp_sock, sock_async_flags_t flags, void *ctx)
+{
+    sock_dtls_t *sock = ctx;
+
+    if (flags & SOCK_ASYNC_MSG_RECV) {
+        sock_dtls_session_t remote;
+        void *data = NULL;
+        void *data_ctx = NULL;
+
+        ssize_t res = sock_udp_recv_buf(udp_sock, &data, &data_ctx, 0,
+                                        &remote.ep);
+        if (res <= 0) {
+            DEBUG("sock_dtls: error receiving UDP packet: %d\n", (int)res);
+            return;
+        }
+
+        /* prevent overriding already set `buf_ctx` */
+        if (sock->buf_ctx != NULL) {
+            DEBUG("sock_dtls: unable to store buffer asynchronously\n");
+            _check_more_chunks(udp_sock, &data, &data_ctx, &remote.ep);
+            return;
+        }
+        _ep_to_session(&remote.ep, &remote.dtls_session);
+        sock->buf_ctx = data_ctx;
+        res = dtls_handle_message(sock->dtls_ctx, &remote.dtls_session,
+                                  data, res);
+        if (sock->buf == NULL) {
+            _check_more_chunks(udp_sock, &data, &data_ctx, &remote.ep);
+            sock->buf_ctx = NULL;
+        }
+    }
+    if ((flags & SOCK_ASYNC_PATH_PROP) && sock->async_cb) {
+        /* just hand this event type up the stack */
+        sock->async_cb(sock, SOCK_ASYNC_PATH_PROP, sock->async_cb_arg);
+    }
+}
+
+#include "net/sock/async/event.h"
```

Comment:
Why not move this with the other includes?

Reply:
Not sure, will see if it is still compilable when on top (with `sock`/`sock_async` such things can be the reason for a late `#include`).

Reply:
Done.